index.js

```diff
@@ -0,0 +1,346 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+const async = require('async')
+const { Worker } = require('worker_threads')
+const os = require('os')
+
+const Database = require('better-sqlite3')
+const Base = require('bfx-facs-base')
+
+const DB_WORKER_ACTIONS = require(
+  './worker/db-worker-actions/db-worker-actions.const'
+)
+
+class Sqlite extends Base {
+  constructor (caller, opts, ctx) {
+    super(caller, opts, ctx)
+
+    this._hasConf = false
+    this._workers = new Map()
+    this._queue = []
+
+    this.name = 'db-sqlite'
+    this.opts = this._getOpts()
+
+    this.init()
+  }
+
+  _getOpts () {
+    const {
+      unacceptableWalFileSize = 100000000,
+      minWorkersCount = 4,
+      maxWorkersCount = 16,
+      workerPathAbsolute,
+      dbPathAbsolute,
+      name,
+      label
+    } = this.opts
+
+    const _name = name ? `_${name}` : ''
+    const _label = label ? `_${label}` : ''
+    const baseName = `${this.name}${_name}${_label}.db`
+
+    const dbPath = (
+      typeof dbPathAbsolute === 'string' &&
+      path.isAbsolute(dbPathAbsolute)
+    )
+      ? path.join(dbPathAbsolute, baseName)
+      : path.join(this.caller.ctx.root, 'db', baseName)
+    const workerPath = (
+      typeof workerPathAbsolute === 'string' &&
+      path.isAbsolute(workerPathAbsolute)
+    )
+      ? workerPathAbsolute
+      : path.join(__dirname, 'worker', 'index.js')
+
+    return {
+      ...this.opts,
+      dbPath,
+      workerPath,
+      minWorkersCount,
+      maxWorkersCount,
+      unacceptableWalFileSize
+    }
+  }
+
+  _initDb (cb) {
+    const {
+      isSqliteStoredInMemory,
+      isNotWorkerSpawned,
```

Comment:
naming: i don't understand the setting by its name. it the correct name: "noWorkerThreads"?

Reply:
yes, it's the setting to avoid spawning any worker threads
would change to that naming